streaming_providers/realdebrid/utils.py

```diff
@@ -46,21 +46,16 @@ def get_video_url_from_realdebrid(
             raise ProviderException(
                 "Torrent limit reached. Please try again later.", "torrent_limit.mp4"
             )
-        if info_hash in response["list"]:
+
+        torrent_id = rd_client.add_magnet_link(magnet_link).get("id")
+        if not torrent_id:
             raise ProviderException(
-                "Torrent is already being downloading", "torrent_not_downloaded.mp4"
+                "Failed to add magnet link to Real-Debrid", "transfer_error.mp4"
             )
-
-        torrent_id = rd_client.add_magent_link(magnet_link).get("id")
         torrent_info = rd_client.get_torrent_info(torrent_id)
     else:
         torrent_id = torrent_info.get("id")
 
-    if not torrent_id:
```

Comment:
What if the else condition 
```
torrent_id = torrent_info.get("id")
```

return None?

Reply:
As mentioned in above comment, my understanding is as follows:

1. `get_available_torrent` returns a valid `torrent_info` if the torrent is available in any state in RD.
2. If `torrent_info` is valid, we can get the `torrent_id`. In case we do get none, I guess we raise an API error.